patches/tModLoader/Terraria/Main.cs.patch

```diff
@@ -1068,15 +1068,23 @@
  			if (!dedServ && (double)screenPosition.Y < worldSurface * 16.0 + 16.0 && netMode != 2) {
  				Star.UpdateStars();
  				Cloud.UpdateClouds();
-@@ -12806,6 +_,8 @@
+@@ -12806,8 +_,16 @@
  			if (InGameUI != null)
  				InGameUI.Update(gameTime);
  
 +			ModHooks.UpdateUI(gameTime);
 +			PlayerInput.ScrollWheelDeltaForUI = 0; //TODO: Should this be before PlayerInput.ResetInputsOnActiveStateChange()?
  			CreativeMenu.Update(gameTime);
++
++			if (gameMenu) //Don't update IBigProgressBars in the main menu to avoid IOOB during unload when modded NPCs were alive --direwolf420
++				goto afterBigBossProgressBarUpdate; //Avoids indentation change
```

Comment:
Why avoid changing the indentation in original code? Does it mess with the patching in some way?

Reply:
In this case not really actually, its more used for larger chunks of code to prevent exposing vanilla code too much (indentation change is picked up by the patcher).

Reply:
Sure, I can understand that. Generally speaking, aren't the original developers in favor of modding in this game?

Reply:
The primary reason for reducing patch size is for maintainability. When the base game code changes (or is decompiled differently) patches break and become hard to apply if there's lots of changed lines which don't actually have meaningful changes in them.

See <https://github.com/tModLoader/tModLoader/wiki/tModLoader-Style-Guide> for some guidelines

Reply:
Honestly though @direwolf420 as this is a _one line_ indentation change. I'd rather you keep the indentation change. It's about keeping patch size _small_ primarily. I'll merge it after that adjustment

Reply:
done

Reply:
I think you still need to invert the condition? Or am I missing something?

Reply:
I did the indentation skip thing after having the correct condition, fixed the condition, then undid the indentation, and forgot to fix the condition again lol